server/src/main.rs

```diff
@@ -1,19 +1,46 @@
 mod routes;
 
-use std::io;
+use std::{env, io};
 
 use actix_files::Files;
-use actix_web::{middleware::Logger, App, HttpServer};
+use actix_session::{storage::CookieSessionStore, SessionMiddleware};
+use actix_web::{cookie::Key, middleware::Logger, web, App, HttpServer};
+use dotenv::dotenv;
 
+/// Returns a secret key that is used to encrypt the session cookie stored by the client.
+///
+/// # Panics
+///
+/// Panics if the environment variable `SESSION_SECRET` is not a base64-encoded string of at least
+/// 32 randomly-generated bytes.
+fn get_session_secret() -> Key {
+    let session_secret =
+        env::var("SESSION_SECRET").expect("environment variable `SESSION_SECRET` must be defined");
+    let session_secret = base64::decode(session_secret)
+        .expect("cannot decode environment variable `SESSION_SECRET` from base64");
+    Key::derive_from(&session_secret)
+}
+
+/// Launches the web server.
 #[actix_web::main]
 async fn main() -> io::Result<()> {
+    dotenv().ok();
+
     env_logger::init();
 
     HttpServer::new(|| {
         App::new()
             .wrap(Logger::default())
-            .service(routes::login::handler)
-            .service(routes::logout::handler)
+            .wrap(
+                SessionMiddleware::builder(CookieSessionStore::default(), get_session_secret())
+                    .cookie_name("session".to_string())
+                    .build(),
+            )
+            .service(
+                web::scope("/api")
```

Comment:
Why `/api`?

Reply:
I'm planning for all back-end server routes to be prefixed with `/api`. This is in contrast to all SPA routes, which will just contain the user-friendly name (e.g. `/login`, which is handled by the React router). This makes it clear which routes are handled by the front end and which are handled by the back end.

Reply:
API (back-end) routes will also interface using `Content-Type: application/json`, rather than serve HTML content. These routes are not meant to be viewable by the user, they are only meant for the front end to make `fetch` calls in the background.